src/tasks/dailyReminder.ts

```diff
@@ -1,24 +1,39 @@
 import { Task } from 'klasa';
 
-import { PerkTier, Time } from '../lib/constants';
+import { client } from '../index';
+import { Time } from '../lib/constants';
 import { UserSettings } from '../lib/settings/types/UserSettings';
 import { noOp } from '../lib/util';
-import getUsersPerkTier from '../lib/util/getUsersPerkTier';
+
+const DailyInterval = 12 * 60 * 60;
+const DailyTickInterval = Time.Second * 60;
 
 export default class extends Task {
+	async init() {
+		if (this.client.dailyReminderTicker) {
+			clearInterval(this.client.dailyReminderTicker);
+		}
+		this.client.dailyReminderTicker = setInterval(this.dailyReminderTick.bind(this), DailyTickInterval);
+	}
+
 	async run() {
+		this.dailyReminderTick();
+	}
+
+	async dailyReminderTick() {
 		const currentDate = Date.now();
 
-		for (const user of this.client.users.cache.values()) {
-			if (getUsersPerkTier(user) < PerkTier.Two) continue;
-			const lastVoteDate = user.settings.get(UserSettings.LastDailyTimestamp);
-			if (lastVoteDate === -1) continue;
+		const dailyReady = currentDate - DailyInterval;
+		const result = await client.query<{ id: string }[]>(
+			`SELECT id FROM users WHERE bitfield && '{2,3,4,5,6}'::int[] AND "lastDailyTimestamp" != -1 AND "lastDailyTimestamp" < ${dailyReady};`
+		);
+
+		if (result.length === 0) return null;
+		for (const userID of result.values()) {
```

Comment:
`userID.id` doesnt make sense.

Reply:
you dont need to check if the length is 0, if theres no results, it will just have nothing to iterate over and theres nothing wrong with that.